spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -10,10 +18,34 @@
     end
 
     context "when sending a welcome_thread notification" do
-      xit "generates the appropriate broadcast to be sent to a user"
-      xit "it sends a welcome notification for that broadcast"
-      xit "it does not send duplicate welcome notification for that broadcast"
-      xit "does not send a notification to a user who has commented in a welcome thread"
+      let(:welcome_thread_article) { create(:article, title: "Welcome Thread - v3000", published: true) }
+      let!(:welcome_thread_comment) { create(:comment, commentable_id: welcome_thread_article.id, commentable_type: "Article", user_id: user.id) }
+
+      it "generates the correct broadcast type and sends the notification to the user", :aggregate_failures do
+        expect(receiving_user.notifications.count).to eq(0)
+        sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
+
+        expect(receiving_user.notifications.count).to eq(1)
+        expect(receiving_user.notifications.first.notifiable).to eq(welcome_broadcast)
+      end
+
+      it "does not send a notification to a user who has commented in a welcome thread", :aggregate_failures do
+        expect(welcome_thread_article.comments.count).to eq(1)
+        expect(welcome_thread_comment.commentable).to eq(welcome_thread_article)
+        expect(user.comments.count).to eq(1)
```

Comment:
If you put the work of assigning the welcome thread article + comments into a `before` block, I think it would more obvious to quickly see what you are testing, and then I don't think you need all of these assertions.

Effectively, what lines 33-35 are doing here is just testing that you set up things correctly in lines 21-22. 🙃 

Instead, the test probably could be trimmed down a bit to just test the actual behavior that should or should not change when you run the generator (`described_class.call(...)`).

Something like this:
```ruby
it "does not send a notification to a user who has a comment on the welcome thread" do
  expect(user.notifications.count).to eq(0)
  sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
  expect(user.notifications.count).to eq(0)
end
```
tells us pretty clearly what we expect to happen when we try to call the generator on a user who has already commented. We can see from the expectation that nothing should change, and they shouldn't see any new notifications.

You might also be able to write it like this, too:
```ruby
it "does not send a notification to a user who has a comment on the welcome thread" do
  expect do
    sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
  end.not_to change(user_with_comment.notifications, :count)
end
```

Right now, you're testing a bunch of assertions (the comments count on the article, the commentable on the comment, the comments on the user) before running the actual behavior of the test, but then you don't actually do anything with those assertions/don't test if that they have changed or not after running `described_class.call(...)`. This is probably a pretty good indicator that the expectations on lines 33-35 don't really have anything to do with the behavior of the test itself; they're testing the setup for the test, instead.

Reply:
Makes sense to me! I've made this change.